lib/bulk_insert/statement_adapters/generic_adapter.rb

```diff
@@ -0,0 +1,17 @@
+module BulkInsert
+  module StatementAdapters
+    class GenericAdapter
```

Comment:
I considered GenericAdapter as a some kind of fallback for other databases. 

Adding `BaseAdapter` or `AbstractAdapter` is a good idea though.

Reply:
Overall the approach looks very good to me. You may want to take in account this change before merging. You may include in the base class also the rest of the common methods to have a complete adapter (if anything else can be extracted). This will make easier the extension creating a new adapter

Reply:
I've added base adapter here: 1dcc5e3ca7df269849cc3008f9e07f339b4222e4

> You may include in the base class also the rest of the common methods (...)

I prefer to follow YAGNI rule here, so instead of forward-thinking what could be extracted, let's just keep what's needed now and extract additional methods when/if needed in the future. Also, I don't want this PR and amount of changes to grow too much ;-)